backend/dataall/modules/dataset_sharing/aws/glue_client.py

```diff
@@ -130,3 +132,38 @@ def delete_database(self):
                 f'due to: {e}'
             )
             raise e
+
+    def get_source_catalog(self):
+        """ Get the source catalog account details """
+        try:
+            log.info(f'Fetching source catalog details for database {self._database}...')
+            response = self._client.get_database(CatalogId=self._account_id, Name=self._database)
+            linked_database = response.get('Database', {}).get('TargetDatabase', {})
+            log.info(f'Fetched source catalog details for database {self._database} are: {linked_database}...')
+            if linked_database:
+                return Catalog(account_id=linked_database.get('CatalogId'),
+                               database_name=linked_database.get('DatabaseName'),
+                               region=linked_database.get('Region', self._region))
+        except Exception as e:
+            log.exception(f'Could not fetch source catalog details for database {self._database} due to {e}')
+            raise e
```

Comment:
Im actually not sure if we should even return a None here? If we can't get catalog details we should probably just fail?

Reply:
When returning None, the source aws account Id will be the source environment accounts id and not catalog account id. "None" is returned when the database is not in the catalog account